tests/integration/test_metadata_model.py

```diff
@@ -54,14 +72,20 @@ def test_submit_filebased_manifest(
         # GIVEN a metadata model object using class labels
         meta_data_model = metadata_model(helpers, "class_label")
 
-        # AND a filebased test manifset
-        manifest_path = helpers.get_data_path(manifest_path)
+        # AND a filebased test manifest
+        load_args = {
+            "dtype": "string",
+        }
+        manifest = helpers.get_data_frame(
+            manifest_path, preserve_raw_input=False, **load_args
+        )
+        manifest_full_path = helpers.get_data_path(manifest_path)
 
-        # WHEN the manifest it submitted
+        # WHEN the manifest it submitted and files are annotated
         # THEN submission should complete without error
         with does_not_raise():
             manifest_id = meta_data_model.submit_metadata_manifest(
-                manifest_path=manifest_path,
+                manifest_path=manifest_full_path,
                 dataset_id=dataset_id,
                 manifest_record_type="file_and_entities",
```

Comment:
I am only seeing `manifest_record_type = "files_and_entities"` got tested. Can we also have `manifest_record_type="table_and_file"` got tested somewhere too? 

when submitting a manifest with parameter`manifest_record_type="table_and_file"`, this function `upload_manifest_as_table` (see [here](https://github.com/Sage-Bionetworks/schematic/blob/develop/schematic/store/synapse.py#L1946)) would get triggered. And this function would do the following in terms of adding annotations: 
* adding annotations to entities listed in the manifest (see [here](https://github.com/Sage-Bionetworks/schematic/blob/develop/schematic/store/synapse.py#L1997C22-L1997C55)) 
* setting annotations for the file itself (see [here](https://github.com/Sage-Bionetworks/schematic/blob/develop/schematic/store/synapse.py#L2017)) 
* setting annotations for the table (see [here](https://github.com/Sage-Bionetworks/schematic/blob/develop/schematic/store/synapse.py#L2017))

From my perspective, ideally the test case should test that annotations are added for all these three cases. Can we have this added in the PR, if not, can we open a new ticket? 

Also, when `manifest_record_type = "files_and_entities"`, when submitting a manifest, this function [here](https://github.com/Sage-Bionetworks/schematic/blob/develop/schematic/store/synapse.py#L2042) got called. And this function would do the following in terms of adding annotations: 
* adding annotations to entities listed in the manifest (see [here](https://github.com/Sage-Bionetworks/schematic/blob/develop/schematic/store/synapse.py#L2073))
* setting annotations for the file itself (see [here](https://github.com/Sage-Bionetworks/schematic/blob/develop/schematic/store/synapse.py#L2093))

By reading your test, I am confident that bullet point 1 is tested, but bullet point 2 is not. Can we add test for bullet point 2?

Reply:
Hi Jenny! Sorry for the late reply. I was out yesterday. Your understanding that each manifest file could only contain one component type is correct 👍 . 

For your second question, I just noticed that we don't have a unit test for it (which I just create a ticket for it: https://sagebionetworks.jira.com/browse/FDS-2421). From my understanding, this function `format_manifest_annotations` handles adding annotations for manifest file itself and also the table. Here's how it works: 
* when manifest_record_type = "file_only", The code (see the "if" statement there) in `format_manifest_annotation` makes sure that component in the manifest gets captured in the annotations. In this case, Component = `BulkRNA-seqAssay` got added as an annotation key pair. As you could see below, the manifest csv got annotated like below eventually: 
<img width="1043" alt="Screenshot 2024-09-17 at 3 33 10 PM" src="https://github.com/user-attachments/assets/ead08241-9ab6-4b91-93df-dbd1c2fa61c5">


* when manifest_record_type = "table_and_file", we add annotations to table: 
<img width="561" alt="Screenshot 2024-09-17 at 3 38 14 PM" src="https://github.com/user-attachments/assets/f800df27-fc1a-4bc6-ab60-6f6dc41e0268">
But as you could see, it only contains etag and id

Reply:
@linglp thank you for the breakdown, that makes sense. It looks like [the logic of adding the `Component` annotation](https://github.com/Sage-Bionetworks/schematic/blob/5721c77d35ce14b45c7bf768af133c6d81eb4abe/schematic/store/synapse.py#L1618) is only triggered under `is_file` and not `is_table`, which is why there's no `Component` annotation in the tables. 

What is the purpose of adding this `Component` annotation to the manifest file itself? My first assumption would be: because it makes it easier to see what model is being used to annotate the entities, but then that raises the question of: why are the tables not annotated the same way?